rotkehlchen/tests/unit/test_protocol_balances.py

```diff
@@ -969,3 +977,46 @@ def test_extrafi_farm_balances(
             usd_value=FVal('10012.6978598973939729841870'),
         )},
     )}
+
+
+@pytest.mark.vcr(filter_query_parameters=['apikey'])
+def test_extrafi_cache(optimism_inquirer: 'OptimismInquirer'):
+    """Check that cache gets populated and timestamp update if
+    we requery again"""
+    with GlobalDBHandler().conn.write_ctx() as write_cursor:
+        write_cursor.execute(
+            'DELETE FROM general_cache WHERE key LIKE "EXTRAFI%"',
+        )
+        write_cursor.execute(
+            'DELETE FROM unique_cache WHERE key LIKE "EXTRAFI%"',
+        )
+
+    query_extrafi_data(
+        inquirer=optimism_inquirer,
+        cache_type=CacheType.EXTRAFI_NEXT_RESERVE_ID,
+        msg_aggregator=optimism_inquirer.database.msg_aggregator,
+    )
+    chain = str(optimism_inquirer.chain_id.serialize_for_db())
+    with GlobalDBHandler().conn.read_ctx() as cursor:
+        existing_pools = get_existing_reward_pools(chain_id=ChainID.OPTIMISM)
+        assert '0x1A6cB72ba9aD6Cf0855D5B497e8E1485A3B3FE39' in existing_pools[0]
+        last_queried_ts = globaldb_get_unique_cache_last_queried_ts_by_key(
+            cursor=cursor,
+            key_parts=(CacheType.EXTRAFI_NEXT_RESERVE_ID, chain),
+        )
+        assert should_update_protocol_cache(
+            cache_key=CacheType.EXTRAFI_NEXT_RESERVE_ID,
+            args=(chain,),
+        ) is False
+
+    time.sleep(1)  # give some time to avoid querying to fast and see the timestamp modified
```

Comment:
1. typo
2. Why? `time.sleep()` is generally a bad idea in tests and anywhere. Shouldn't you instead use freezetime and move the current timestamp? This should work too and not depend on randomness like time.sleep. Also is probably faster.

```suggestion
    time.sleep(1)  # give some time to avoid querying too fast and see the timestamp modified
```